diff -Nur ccextractor.0.80/linux/Makefile ccextractor.0.80.msys2/linux/Makefile
--- ccextractor.0.80/linux/Makefile	2016-02-17 11:46:49.000000000 +0100
+++ ccextractor.0.80.msys2/linux/Makefile	2016-05-28 17:24:11.128675700 +0200
@@ -6,7 +6,7 @@
 INCLUDE = -I../src/gpacmp4/ -I../src/libpng -I../src/zlib -I../src/lib_ccx -I../src/.
 INCLUDE += -I../src/zvbi
 ALL_FLAGS = -Wno-write-strings -D_FILE_OFFSET_BITS=64
-LDFLAGS = -lm
+LDFLAGS = -static -lm -lwinmm -lws2_32
 
 ifneq (, $(findstring linux, $(SYS)))
 CFLAGS +=-DGPAC_CONFIG_LINUX
@@ -14,7 +14,7 @@
 TARGET = ccextractor
 
 OBJS_DIR = objs
-VPATH = ../src:../src/gpacmp4:../src/libpng:../src/zlib:../src/lib_ccx:../src/zvbi
+VPATH = ../src:../src/gpacmp4:../src/libpng:../src/zlib:../src/lib_ccx:../src/zvbi:../src/win_iconv
 
 SRCS_DIR = ../src
 SRCS_C = $(wildcard $(SRCS_DIR)/*.c)
@@ -42,6 +42,10 @@
 SRCS_ZLIB = $(wildcard $(SRCS_ZLIB_DIR)/*.c)
 OBJS_ZLIB = $(SRCS_ZLIB:$(SRCS_ZLIB_DIR)/%.c=$(OBJS_DIR)/%.o)
 
+SRCS_ICONV_DIR = $(SRCS_DIR)/win_iconv
+SRCS_ICONV = $(wildcard $(SRCS_ICONV_DIR)/*.c)
+OBJS_ICONV = $(SRCS_ICONV:$(SRCS_ICONV_DIR)/%.c=$(OBJS_DIR)/%.o)
+
 INSTLALL = cp -f -p
 INSTLALL_PROGRAM = $(INSTLALL)
 DESTDIR = /usr/bin
@@ -89,8 +93,8 @@
 objs_dir: 
 	mkdir -p $(OBJS_DIR)
 
-$(TARGET): $(OBJS) $(OBJS_PNG) $(OBJS_GPACMP4) $(OBJS_ZVBI) $(OBJS_ZLIB) $(OBJS_CCX)
-	$(CC) $(ALL_FLAGS) $(CFLAGS) $(OBJS) $(OBJS_CCX) $(OBJS_PNG) $(OBJS_ZVBI) $(OBJS_GPACMP4) $(OBJS_ZLIB) $(LDFLAGS) -o $@
+$(TARGET): $(OBJS) $(OBJS_PNG) $(OBJS_GPACMP4) $(OBJS_ZVBI) $(OBJS_ZLIB) $(OBJS_ICONV) $(OBJS_CCX)
+	$(CC) $(ALL_FLAGS) $(CFLAGS) $(OBJS) $(OBJS_CCX) $(OBJS_PNG) $(OBJS_ZVBI) $(OBJS_GPACMP4) $(OBJS_ZLIB) $(OBJS_ICONV) $(LDFLAGS) -o $@
 
 $(OBJS_DIR)/%.o: %.c
 	$(CC) -c $(ALL_FLAGS) $(INCLUDE) $(CFLAGS) $< -o $@  
@@ -104,7 +108,7 @@
 .PHONY: clean
 clean:
 	rm -rf $(TARGET) 2>/dev/null || true
-	rm -rf $(OBJS_CCX) $(OBJS_PNG) $(OBJS_ZLIB) $(OBJS_GPACMP4) $(OBJS) 2>/dev/null || true
+	rm -rf $(OBJS_CCX) $(OBJS_PNG) $(OBJS_ZLIB) $(OBJS_ICONV) $(OBJS_GPACMP4) $(OBJS) 2>/dev/null || true
 	rm -rdf $(OBJS_DIR) 2>/dev/null || true
 	rm -rf .depend 2>/dev/null || true
 
diff -Nur ccextractor.0.80/src/ccextractor.c ccextractor.0.80.msys2/src/ccextractor.c
--- ccextractor.0.80/src/ccextractor.c	2016-03-18 09:51:37.000000000 +0100
+++ ccextractor.0.80.msys2/src/ccextractor.c	2016-05-28 17:26:44.031215200 +0200
@@ -316,7 +316,9 @@
 			s1, s2);
 	}
 #endif
+	/* SEGFAULT with -out=report
 	dbg_print(CCX_DMT_708, "[CEA-708] The 708 decoder was reset [%d] times.\n", ctx->freport.data_from_708->reset_count);
+	*/
 
 	if (is_decoder_processed_enough(ctx) == CCX_TRUE)
 	{
diff -Nur ccextractor.0.80/src/lib_ccx/ccx_common_platform.h ccextractor.0.80.msys2/src/lib_ccx/ccx_common_platform.h
--- ccextractor.0.80/src/lib_ccx/ccx_common_platform.h	2015-07-14 09:28:55.000000000 +0200
+++ ccextractor.0.80.msys2/src/lib_ccx/ccx_common_platform.h	2016-05-28 17:25:18.842525600 +0200
@@ -23,9 +23,13 @@
 		#define STDOUT_FILENO 1
 		#define STDERR_FILENO 2
 		#include "inttypes.h"
+#ifndef UINT64_MAX
 		#define UINT64_MAX   _UI64_MAX
+#endif
 		typedef int socklen_t;
+#ifndef UINT64_MAX
 		typedef int ssize_t;
+#endif
 		typedef uint32_t in_addr_t;
 		#ifndef IN_CLASSD
 			#define IN_CLASSD(i)       (((INT32)(i) & 0xf0000000) == 0xe0000000)
